src/Engine/babylon.engine.ts

```diff
@@ -3537,7 +3537,16 @@
             var linked = context.getProgramParameter(shaderProgram, context.LINK_STATUS);
 
             if (!linked) {
-                context.validateProgram(shaderProgram);
+                var error = context.getProgramInfoLog(shaderProgram);
+                if (error) {
+                    throw new Error(error);
+                }
+            }
+
+            context.validateProgram(shaderProgram);
+            var validated = context.getProgramParameter(shaderProgram, context.LINK_STATUS);

```

Comment:
why doing it twice?

Reply:
The first time, you get the link status, the second time, the validation status. Both might output different messages.

Reply:
oh, I see the mistake here, it should be context.VALIDATE_STATUS